Support for Numpy BitGenerators PR#2: Standard Distributions support (#8038)
numba/numba: docs/source/reference/numpysupported.rst

``random``
----------

Generator Objects
'''''''''''''''''

> Numba supports ``numpy.random.Generator()`` objects. Users can now pass

Reviewer: If this is processed by Sphinx, then double backticks will lack the formatting (and sometimes linking) provided by single ticks and a domain/role specification.
E.g.
```suggestion
Numba supports :py:class:`numpy.random.Generator` objects. Users can now pass
```

The following :py:class:`Generator` methods are supported:

* :func:`numpy.random.Generator().exponential()`
* :func:`numpy.random.Generator().gamma()` (*)
* :func:`numpy.random.Generator().normal()` (*)
* :func:`numpy.random.Generator().random()`
* :func:`numpy.random.Generator().standard_normal()`
* :func:`numpy.random.Generator().standard_exponential()`
* :func:`numpy.random.Generator().standard_gamma()` (*)
* :func:`numpy.random.Generator().uniform()` (*)

.. note::
  Users can expect Numba to be bit accurate with Numpy within

>   the last few (at max 5) ULPs for Linux, Windows and macOS.

Reviewer 2: ```suggestion
  the last few (at max 5) ULPs for Linux-x86_64, Windows and macOS.
```
```suggestion
  the last few (at max 5) ULPs for Linux, Windows and macOS.
```

>   the last few (at max 5) ULPs for Linux-x86_64, Windows and macOS.

Reviewer 3: ```suggestion
  the last few (at max 5) ULPs for Linux-x86_64, Windows-x86_64 and macOS.
```

  For distributions marked above with `(*)`: Due to certain compiler
  technicalities, there might be discrepanies of order of 1000 ULPs

>   on 32-bit, Aarch and ppc64le distributions for Linux.

Reviewer 2: ```suggestion
  on 32-bit architectures as well as linux-aarch64 and linux-ppc64le platforms.
```
```suggestion
  on 32-bit, Aarch and ppc64le distributions for Linux.
```

>   For distributions marked above with `(*)`: Due to certain compiler
>   technicalities, there might be discrepanies of order of 1000 ULPs
>   on 32-bit architectures as well as linux-aarch64 and linux-ppc64le platforms.
>   For more details, `see the comment here <https://github.com/numba/numba/pull/8038#issuecomment-1165571368>`

Reviewer 3: ```suggestion
  For distributions marked above with `(*)`: Due to instruction selection
  differences across compilers, there may be discrepancies in the order of
  1000s of ULPs on 32-bit architectures as well as linux-aarch64 and
  linux-ppc64le platforms. The differences are unlikely to impact the "quality"
  of the random number generation as they occur through changes in
  rounding that happen when fused-multiply-add is used instead of multiplication
  followed by addition.
```
suggest noting the impact on the RNG quality and offering a brief explanation to users.

The following :py:class:`Generator` methods are supported:

* :func:`numpy.random.Generator().exponential()`
* :func:`numpy.random.Generator().gamma()` (*)
* :func:`numpy.random.Generator().normal()` (*)
* :func:`numpy.random.Generator().random()`
* :func:`numpy.random.Generator().standard_normal()`
* :func:`numpy.random.Generator().standard_exponential()`
* :func:`numpy.random.Generator().standard_gamma()` (*)
* :func:`numpy.random.Generator().uniform()` (*)

.. note::

>   Users can expect Numba replicate NumPy's results to within

Reviewer 3: ```suggestion
  Users can expect Numba to replicate NumPy's results to within
```
